x-pack/plugins/spaces/public/plugin.test.ts

```diff
@@ -209,27 +209,156 @@ describe('Spaces plugin', () => {
-    const coreSetup = coreMock.createSetup();
-    const coreStart = coreMock.createStart();
+  describe('hasOnlyDefaultSpace', () => {
+    it('determines hasOnlyDefaultSpace correctly when maxSpaces=1', () => {
+      const coreSetup = coreMock.createSetup();
+      const coreStart = coreMock.createStart();
 
-    const plugin = new SpacesPlugin(coreMock.createPluginInitializerContext({ maxSpaces: 1 }));
-    const spacesSetup = plugin.setup(coreSetup, {});
-    const spacesStart = plugin.start(coreStart);
+      const plugin = new SpacesPlugin(coreMock.createPluginInitializerContext({ maxSpaces: 1 }));
+      const spacesSetup = plugin.setup(coreSetup, {});
+      const spacesStart = plugin.start(coreStart);
 
-    expect(spacesSetup.hasOnlyDefaultSpace).toBe(true);
-    expect(spacesStart.hasOnlyDefaultSpace).toBe(true);
+      expect(spacesSetup.hasOnlyDefaultSpace).toBe(true);
+      expect(spacesStart.hasOnlyDefaultSpace).toBe(true);
+    });
+
+    it('determines hasOnlyDefaultSpace correctly when maxSpaces=1000', () => {
+      const coreSetup = coreMock.createSetup();
+      const coreStart = coreMock.createStart();
+
+      const plugin = new SpacesPlugin(coreMock.createPluginInitializerContext({ maxSpaces: 1000 }));
+      const spacesSetup = plugin.setup(coreSetup, {});
+      const spacesStart = plugin.start(coreStart);
+
+      expect(spacesSetup.hasOnlyDefaultSpace).toBe(false);
+      expect(spacesStart.hasOnlyDefaultSpace).toBe(false);
+    });
   });
 
-  it('determines hasOnlyDefaultSpace correctly when maxSpaces=1000', () => {
-    const coreSetup = coreMock.createSetup();
-    const coreStart = coreMock.createStart();
+  describe('isSolutionViewEnabled', () => {
+    it('when onCloud, not serverless and allowSolutionVisibility is "true"', () => {
+      const coreSetup = coreMock.createSetup();
+      const coreStart = coreMock.createStart();
+      const cloud = cloudMock.createSetup();
+      cloud.isCloudEnabled = true;
+
+      const plugin = new SpacesPlugin(
+        coreMock.createPluginInitializerContext(
+          { allowSolutionVisibility: true },
+          { buildFlavor: 'traditional' }
+        )
+      );
+      const spacesSetup = plugin.setup(coreSetup, { cloud });
+      const spacesStart = plugin.start(coreStart);
+
+      expect(spacesSetup.isSolutionViewEnabled).toBe(true);
+      expect(spacesStart.isSolutionViewEnabled).toBe(true);
+    });
 
-    const plugin = new SpacesPlugin(coreMock.createPluginInitializerContext({ maxSpaces: 1000 }));
-    const spacesSetup = plugin.setup(coreSetup, {});
-    const spacesStart = plugin.start(coreStart);
+    it('when not onCloud and allowSolutionVisibility is "true"', () => {
+      const coreSetup = coreMock.createSetup();
+      const coreStart = coreMock.createStart();
 
-    expect(spacesSetup.hasOnlyDefaultSpace).toBe(false);
-    expect(spacesStart.hasOnlyDefaultSpace).toBe(false);
+      {
```

Comment:
__note:__ `{ ... }` - Great way to scope and reuse `const` values in the same test - never thought of this before. Nice!